compiler/src/dmd/root/array.d

```diff
@@ -41,15 +41,15 @@ public:
      * Params:
      *  dim = initial length of array
      */
-    this(size_t dim) pure nothrow
```

Comment:
Why?

Reply:
because originally I did not fake purity for the allocation functions.
now those changes can go away

Reply:
Ok, if I try adding back the pure qualifiers?